### DISCUSSION TITLE
vscode remote ssh | Copilot could not connect to server

### DISCUSSION THREAD

**Discussion by lgonzalezsa** [community/community#7752] [Copilot Conversations]
Majority of the time am using remote ssh to code, installed copilot on ssh but failed to activate:

`GitHub Copilot could not connect to server. Extension activation failed: "Client network socket disconnected before secure TLS connection was established"`

Any recommendation?

**Comment by MrGolden1** [comment]
Has your ssh server internet connection? Because in my case that I ssh to a Raspberry Pi via LAN, there is no internet connection.

**Reply by lgonzalezsa** [reply]
After post from @MrGolden1 I did another test, instead of trying to use vscode remote SSH extension, just using Jupyter in remote mode to start a new Jupyter session to the desired kernel and now I have the benefit of using copilot in my vscode in my laptop but connected to a remote Jupyter kernel.

**Comment by hamidatjajiga** [comment]
I have the same error:

```
GitHub Copilot could not connect to server. Extension activation failed: "Client network socket disconnected before secure TLS connection was established"
```
